slack_bolt/app/app.py

```diff
@@ -348,12 +351,27 @@ def middleware_next():
             self._framework_logger.debug(debug_checking_listener(listener_name))
             if listener.matches(req=req, resp=resp):
                 # run all the middleware attached to this listener first
-                resp, next_was_not_called = listener.run_middleware(req=req, resp=resp)
+                middleware_resp, next_was_not_called = listener.run_middleware(
+                    req=req, resp=resp
+                )
                 if next_was_not_called:
+                    if middleware_resp is not None:
+                        if self._framework_logger.level <= logging.DEBUG:
+                            debug_message = debug_return_listener_middleware_response(
+                                listener_name,
+                                middleware_resp.status,
+                                middleware_resp.body,
+                                starting_time,
+                            )
+                            self._framework_logger.debug(debug_message)
+                        return middleware_resp
                     # The last listener middleware didn't call next() method.
                     # This means the listener is not for this incoming request.
                     continue
 
+                if middleware_resp is not None:
```

Comment:
This would be the case when a listener middleware sets the response _but also_ calls next, right?

Reply:
yes, it is 👍